include/mitsuba/render/shape.h

```diff
@@ -240,7 +245,69 @@ class MTS_EXPORT_RENDER Shape : public Object {
      */
     virtual std::pair<Vector3f, Vector3f> normal_derivative(const SurfaceInteraction3f &si,
                                                             bool shading_frame = true,
-                                                            Mask active        = true) const;
+                                                            Mask active = true) const;
+
+    /**
+     * \brief Evaluate a specific shape attribute at the given surface interaction
```

Comment:
Maybe add a 1-line comment explaining what these generally are. Like ``Shape attributes are user-provided fields that provide extra information at an intersection. An example of this would be a per-vertex or per-face color on a triangle mesh.``.